### PR TITLE
adding object definition handling and identification

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
adding object definition handling and identification

### CODE REVIEW

File: demisto_sdk/commands/common/constants.py [hunk 1]
```diff
@@ -163,7 +165,8 @@ class FileType(Enum):
     CONNECTIONS_DIR,
     GENERIC_FIELDS_DIR,
     GENERIC_TYPES_DIR,
-    GENERIC_MODULES_DIR
+    GENERIC_MODULES_DIR,
+    GENERIC_DEFINITIONS_DIR
```
Comment (Itay4) [inline demisto_sdk/commands/common/constants.py:169]:
```suggestion
    GENERIC_DEFINITIONS_DIR,
```

File: demisto_sdk/commands/common/schemas/genericdefinition.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+type: map
+mapping:
+  id:
+    type: str
```
Comment (Itay4) [inline demisto_sdk/commands/common/schemas/genericdefinition.yml:4]: shouldnt `id` and `name` be required?

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -1099,9 +1099,12 @@ def find_type(path: str = '', _dict=None, file_type: Optional[str] = None, ignor
         if 'group' in _dict and LAYOUT_CONTAINER_FIELDS.intersection(_dict):
             return FileType.LAYOUTS_CONTAINER
 
-        if 'definitions' in _dict and 'views' in _dict:
+        if 'definitionIds' in _dict and 'views' in _dict:
             return FileType.GENERIC_MODULE
 
+        if 'auditable' in _dict:
```
Comment (Itay4) [inline demisto_sdk/commands/common/tools.py:1105]: not sure about this as `auditable` might be an attribute for every entity

File: requirements.txt [hunk 4]
```diff
@@ -3,7 +3,7 @@ autopep8>=1.4.4
 bandit>=1.6.2
 bs4>=0.0.1
 chardet==3.0.2
-click>=8.0.1
+click>=7.1.0, <8.0.0
```
Comment (Itay4) [inline requirements.txt:6]: can we lock the version?

**Review by Itay4 — APPROVED** [review]

File: requirements.txt (hunk 4 above)
Comment (roysagi) [reply requirements.txt]: done

File: TestSuite/pack.py [hunk 5]
```diff
@@ -307,6 +310,15 @@ def create_generic_module(
         self.generic_modules.append(generic_module)
         return generic_module
 
+    def create_generic_definition(
+            self,
+            name,
+            content: dict = None) -> JSONBased:
+        prefix = 'genericdefinition'
+        generic_definition = self._create_json_based(name, prefix, content, dir_path=self._generic_definitions_path)
+        self.generic_modules.append(generic_definition)
```
Comment (sabaghadar) [inline TestSuite/pack.py:319]:
```suggestion
        self.generic_definitions.append(generic_definition)
```

**Pull request merged**